src/main/java/org/symphonyoss/symphony/messageml/elements/UIAction.java

```diff
@@ -0,0 +1,167 @@
+package org.symphonyoss.symphony.messageml.elements;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.symphonyoss.symphony.messageml.MessageMLContext;
+import org.symphonyoss.symphony.messageml.MessageMLParser;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class specify the Symphony Component UIAction which is represented by the tag name "ui-action".
+ * The messageML representation of this element can contain the following fields:
+ * <ul>
+ *   <li>trigger  -> default 'click', trigger of the action. For now only 'click' is supported</li>
```

Comment:
I was wondering if we should enforce that, if we don't it means we can support more features without updating the messageml lib but then validation is less restrictive... I guess it is a balance :)

Reply:
I don't really know how many triggers we are going to support in the future. Or if adding new triggers still would need to allow other attributes. 
Maybe to start we can keep it enforced and then, if we see that the list grows we can remove the default value and simplify the validation.